fail2ban/server/filter.py

```diff
@@ -898,22 +898,18 @@ def textToIp(text, useDns):
 		return ipList
 
 	@staticmethod
-	def cidr(i, n):
-		""" Convert an IP address string with a CIDR mask into a 32-bit
-			integer.
+	def addr2bin(ipstring, cidr=None):
+		""" Convert a string IPv4 address into decimal form.
+		If cidr is supplied, return the network address for the given block
 		"""
-		# 32-bit IPv4 address mask
-		MASK = 0xFFFFFFFFL
-		return ~(MASK >> n) & MASK & DNSUtils.addr2bin(i)
-
-	@staticmethod
-	def addr2bin(string):
-		""" Convert a string IPv4 address into an unsigned integer.
-		"""
-		return struct.unpack("!L", socket.inet_aton(string))[0]
+		if cidr is None:
+			return struct.unpack("!L", socket.inet_aton(ipstring))[0]
```

Comment:
>    Understood. I think I should step back as well.

>    As a note for later:

>    Is the issue the comment? "Convert a string IPv4 address into decimal
>    form" should be "Convert a string IPv4 address into binary form" ?

yeah.  I still think that may be 'numeric' is the most appropriate here,
but heck -- binary it is ;)